### PR TITLE
Applet programming in java

### LINKED ISSUE #2011: Applet programming in java
### Proposed title of article
 Understanding Applet programming in java

### Introduction paragraph (2-3 paragraphs):
An applet is a small java program that is used in internet programming. 

Applets are run mainly using applet viewer or web browsers that mainly supports java. An applet just like any other stand-alone application it performs many functions of which includes performing arithmetic operations, play sounds, accept inputs from the user and also display graphics.

Applets cannot be run independently like stand-alone applications they therefore  use HTML tags to enable them run in a web browser

### Key takeaways: 
By the end of this tutorial the learner should understand:

1. Definition of an applet and write a 'hello world' program using applet.
2. The lifecycle of an applet.
3. Difference between an applet and java stand-alone application
4. How to invoke an applet.
5. How to convert applications to applets.

### References: 
N/A

### PR DESCRIPTION
**Pull request by Paul22691**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2011

### CODE REVIEW

File: content/articles/applet-programming-in-java/index.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ### Prerequisites
```
Comment (WanjaMIKE) [inline content/articles/applet-programming-in-java/index.md:1]: Move this section after the introduction

File: content/articles/applet-programming-in-java/index.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+An applet performs many functions like any application program, which includes performing arithmetic operations, playing sounds, accepts inputs from the user and displaying graphics. A webpage can comprise texts static images and also the java applets. When an applet program runs in a web page it can display moving images, produce sounds, and produce graphics.
+
+To display a java applet program in a web page it needs to be linked to a HTML program.
+
+### Types of applets
+
+Applets are categorized based on how they embed into a web page i.e. Local and Remote applets
+
+    Local applets-is a locally developed and stored applet. A local applet does not need to use an internet connection to be located simply because the directory is in the local system.
```
Comment (WanjaMIKE) [inline content/articles/applet-programming-in-java/index.md:15]: Remove extra space at beginning of the sentence

File: content/articles/applet-programming-in-java/index.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    <applet code = “Myinstitutionapplet.class” width = “320” height = “120”>
+
+    </applet>
+
+    <hr>
+
+    </html>
+```
+### Conclusion
```
Comment (WanjaMIKE) [inline content/articles/applet-programming-in-java/index.md:152]: Add more content in the conclusion

File: content/articles/applet-programming-in-java/index.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+           public void paint (graphics g) 
+
+           {
+
+               g.drawstring (“My institution is one of the best”, 320,120);
+
+           }
+
+        }
```
Comment (WanjaMIKE) [inline content/articles/applet-programming-in-java/index.md:134]: Explain what the above code does

File: content/articles/applet-programming-in-java/index.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+           public void paint (graphics g) 
+
+           {
+
+               g.drawstring (“Hello world”,25,50);
+
+           }
+
+        }
+```
```
Comment (WanjaMIKE) [inline content/articles/applet-programming-in-java/index.md:61]: what does the `paint()` function do. In short, explain the code above

File: content/articles/applet-programming-in-java/index.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+2.  **START/ RUNNING**-this takes place shortly after the applet initialization happens or when the applet was in the idle state. We can call the start state more than once.
+
+3.  **IDLE/ STOPPED**- this result once the applet has stopped running and a halt in the running of an applet may occur once you quit the page where the applet is currently running or when you call the stop () explicitly.
+
+4. **DEAD/ DESTOYED**- this occurs once you quit the web browser from which an applet was running as he applet is shielded in a HTML page or by invoking destroy ()
+
+5.  **DISPLAY/ PAINT**- this occurs once the output operations are to be performed on the screen. it is invoked once the applet once to repaint itself or after the start state start ().
+
+### Differences between an applet and a java stand alone application
```
Comment (WanjaMIKE) [inline content/articles/applet-programming-in-java/index.md:76]: Replace `java` with `Java` in the appropriate areas

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
Please review the above comments

**Comment by Paul22691** [comment]
Hello @WanjaMIKE 
changes requested resolved.

**Comment by WanjaMIKE** [comment]
As you know, we were working on a new structure for our EngEd Repo, which has led to a slight slow down of article publishing. Everything is almost ready now.
In order to publish your article as soon as possible, please do the following:

Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
Delete your article directory from the articles folders.
Do a git pull (take a look at the new structure of the repo after a git pull).
Upload your article files to the new articles folder inside content.
Ping me via Slack if you have any problems.```

**Comment by hectorkambow** [comment]
Thank you v much for your patience as we worked through our Repo re-structuring process. 

The purpose behind this restructure - was to better allow us to handle more volume and allow students to publish more articles and get more involved with the community (with open-source contribution and suggestions). 

With most of the changes coming to completion - we will be sure to get your article published asap. 

Thank you again - and we are looking forward to more amazing articles.

**Comment by Paul22691** [comment]
> As you know, we were working on a new structure for our EngEd Repo, which has led to a slight slow down of article publishing. Everything is almost ready now.
> In order to publish your article as soon as possible, please do the following:
> 
> Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
> Delete your article directory from the articles folders.
> Do a git pull (take a look at the new structure of the repo after a git pull).
> Upload your article files to the new articles folder inside content.
> Ping me via Slack if you have any problems.```



> As you know, we were working on a new structure for our EngEd Repo, which has led to a slight slow down of article publishing. Everything is almost ready now.
> In order to publish your article as soon as possible, please do the following:
> 
> Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
> Delete your article directory from the articles folders.
> Do a git pull (take a look at the new structure of the repo after a git pull).
> Upload your article files to the new articles folder inside content.
> Ping me via Slack if you have any problems.```


@WanjaMIKE already did. Thanks

**Comment by Paul22691** [comment]
> > As you know, we were working on a new structure for our EngEd Repo, which has led to a slight slow down of article publishing. Everything is almost ready now.
> > In order to publish your article as soon as possible, please do the following:
> > Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
> > Delete your article directory from the articles folders.
> > Do a git pull (take a look at the new structure of the repo after a git pull).
> > Upload your article files to the new articles folder inside content.
> > Ping me via Slack if you have any problems.```


@WanjaMIKE already did. Thanks

**Comment by hectorkambow** [comment]
plagiarism check done

**Review by WanjaMIKE — APPROVED** [review]

**Comment by WanjaMIKE** [comment]
@ahmadmardeni1 are you reviewing this order again? Am not in the loop

**Comment by ahmadmardeni1** [comment]
No no am doing the final review @WanjaMIKE 
cc @hectorkambow

**Pull request merged**